src/main/java/bisq/core/dao/governance/merit/MeritConsensus.java

```diff
@@ -117,19 +117,27 @@ public static long getWeightedMeritAmount(long amount, int issuanceHeight, int b
             throw new IllegalArgumentException("amount must not be negative");
         if (blocksPerYear < 0)
             throw new IllegalArgumentException("blocksPerYear must not be negative");
+        if (issuanceHeight < 0)
+            throw new IllegalArgumentException("issuanceHeight must not be negative");
 
 
         // We use a linear function  to apply a factor for the issuance amount of 1 if the issuance was recent and 0
         // if the issuance was 2 years old or older.
+        // To avoid rounding issues with double values we multiply initially with a large number and divide at the end
+        // by that number again. As we multiply the amount in satoshi we get a reasonable good precision even the long
+        // division is not using rounding. Sticking with long values makes that operation more safe against consensus
+        // failures causes by rounding differences with double.
 
-        // TODO Use stairway function without double/divisions or lookup table
-        int maxAge = 2 * blocksPerYear;
-        int age = Math.min(maxAge, blockHeight - issuanceHeight);
+        long maxAge = 2 * blocksPerYear; // maxAge=100 000 (MeritConsensus.BLOCKS_PER_YEAR is 50_000)
+        long age = Math.min(maxAge, blockHeight - issuanceHeight);
+        long inverseAge = maxAge - age;
 
-        double rel = MathUtils.roundDouble((double) age / (double) maxAge, 10);
-        double factor = 1d - rel;
-        long weightedAmount = MathUtils.roundDoubleToLong(amount * factor);
-        log.debug("getWeightedMeritAmount: amount={}, factor={}, weightedAmount={}, ", amount, factor, weightedAmount);
+        // We want a resolution of 1 block so we use the inverseAge and divide by maxAge afterwards to get the
+        // weighted amount
+        // We need to multiply first before we divide!
+        long weightedAmount = (amount * inverseAge) / maxAge;
```

Comment:
I like this better than the previous calculations. Much easier to follow for the same effect.